app/ante/evm/08_vesting.go

```diff
@@ -118,27 +64,30 @@ func CheckVesting(
 	return nil
 }
 
-// UpdateAccountExpenses updates or sets the totalSpend for the given account, then
+// UpdateAccountExpenses updates or sets the expenses for the given account, then
 // returns the new value.
 func UpdateAccountExpenses(
 	ctx sdk.Context,
-	bankKeeper evmtypes.BankKeeper,
+	evmKeeper EVMKeeper,
 	accountExpenses map[string]*EthVestingExpenseTracker,
 	account *vestingtypes.ClawbackVestingAccount,
-	addedExpense *big.Int,
-	denom string,
+	newExpenses *big.Int,
 ) (*EthVestingExpenseTracker, error) {
 	address := account.GetAddress()
 	addrStr := address.String()
 
 	expenses, ok := accountExpenses[addrStr]
 	// if an expense tracker is found for the address, add the expense and return
 	if ok {
-		expenses.Total = expenses.Total.Add(expenses.Total, addedExpense)
+		expenses.Total = expenses.Total.Add(expenses.Total, newExpenses)
 		return expenses, nil
 	}
 
-	balance := bankKeeper.GetBalance(ctx, address, denom)
+	// Get the account balance via EVM Keeper to be sure to have a 18 decimals
+	// representation of the address balance.
+	baseAmount := evmKeeper.GetBalance(ctx, common.BytesToAddress(address.Bytes()))
+	denom := evmtypes.GetEVMCoinDenom()
+	balance := sdk.Coin{Denom: denom, Amount: math.NewIntFromBigInt(baseAmount)}
```

Comment:
I recommend removing this `balance` of type `sdk.Coin`, and work only with the amounts.
It is confusing because it is a balance with the denomination but with the 18 dec amount from the EVM.